api/models/animalSchema.js

```diff
@@ -0,0 +1,98 @@
+const mongoose = require('mongoose');
+const Joi = require('joi');
+
+
+//ANIMAL SCHEMA
+
+const animalSchema = new mongoose.Schema({
+    _id: mongoose.Types.ObjectId,
+    type: {
```

Comment:
Pytanie: czy słowo type nie jest słowem specjalnym? Nie wiem czy to nie będzie powodowało później błędów. Może warto by zmienić np. na animalType?